### ISSUE TITLE
`pyright --verifytypes` doesn't recognize `ExceptionGroup`

### ISSUE THREAD

**Issue by LeeeeT** [microsoft/pyright#4115]
```python
class MyEG(ExceptionGroup[Exception]):
    pass
```

`pyright --verifytypes` produces an error for the above code

```
Symbols used in public interface:
module.MyEG
   error: Type of metaclass unknown
   error: Type of base class unknown
```

**pyright** 1.1.277

**Comment by erictraut** [comment]
`ExceptionGroup` exists only in Python 3.11 and newer. If you configure pyright to use an older version of Python, the symbol `ExceptionGroup` will be unknown because it is defined conditionally in `builtins.pyi`:

```python
if sys.version_info >= (3, 11):
    ...
    class ExceptionGroup(BaseExceptionGroup[_ExceptionT_co], Exception):
        ...
```

You can use the `pythonVersion` command-line argument or configuration setting to specify the python version. If you don't specify one, pyright will use the python version from the currently-selected Python interpreter. If there is no interpreter selected, pyright 1.1.277 will fall back on Python 3.10. Now that Python 3.11 has been released in its final form, I've changed the default to 3.11, so the behavior of pyright 1.1.278 will change.

**Comment by LeeeeT** [comment]
I see this line when I run pyright:
```
Assuming Python version 3.11
```

**Comment by LeeeeT** [comment]
If I add
```toml
[tool.pyright]
pythonVersion = "3.11"
```
to `pyproject.toml`, nothing changes.

**Comment by erictraut** [comment]
Yeah, it looks like `--verifytypes` doesn't use the configured python version. It always uses the default, which is 3.10 for pyright 1.1.277. It should probably be configurable.

**Comment by LeeeeT** [comment]
Thank you!

**Comment by erictraut** [comment]
This is included in pyright 1.1.278, which I just published. It will also be included in a future release of pylance.
